client/src/js/components/bhCurrencySelect.js

```diff
@@ -0,0 +1,98 @@
+angular.module('bhima.components')
+.component('bhCurrencySelect', {
+  controller : bhCurrencySelect,
+  templateUrl : 'partials/templates/bhCurrencySelect.tmpl.html',
+  bindings : {
+    validationTrigger : '<',
+    currencyId : '=',
+    disableIds : '<',
+    onChange : '&'
+  }
+});
+
+bhCurrencySelect.$inject = [ '$scope', 'CurrencyService' ];
```

Comment:
You do realize that `$scope` isn't the entire `$scope` of the page right?  It is an isolated scope.  In this component's case, the only thing attached to the `$scope` is the child form.

I'll be interested to see your design.